src/main/java/org/candlepin/async/tasks/ConsumerMigration.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.async.tasks;
+
+import org.candlepin.config.ConfigProperties;
+import org.candlepin.config.Configuration;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.ContentAccessCertificateCurator;
+import org.candlepin.model.IdentityCertificateCurator;
+import org.candlepin.model.Owner;
+import org.candlepin.model.OwnerCurator;
+
+import com.google.common.collect.Iterables;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+/**
+ * Class responsible for migration of consumers from anonymous orgs.
+ */
+public class ConsumerMigration {
+
+    private static final Logger log = LoggerFactory.getLogger(ConsumerMigration.class);
+
+    private final OwnerCurator ownerCurator;
+    private final ConsumerCurator consumerCurator;
+    private final IdentityCertificateCurator identityCertificateCurator;
+    private final ContentAccessCertificateCurator contentAccessCertificateCurator;
+    private final int batchSize;
+
+    @Inject
+    public ConsumerMigration(OwnerCurator ownerCurator, ConsumerCurator consumerCurator,
+        IdentityCertificateCurator identityCertificateCurator,
+        ContentAccessCertificateCurator contentAccessCertificateCurator,
+        Configuration config) {
+        this.ownerCurator = Objects.requireNonNull(ownerCurator);
+        this.consumerCurator = Objects.requireNonNull(consumerCurator);
+        this.identityCertificateCurator = Objects.requireNonNull(identityCertificateCurator);
+        this.contentAccessCertificateCurator = Objects.requireNonNull(contentAccessCertificateCurator);
+        this.batchSize = config.getInt(ConfigProperties.CONSUMER_MIGRATION_BATCH_SIZE);
+    }
+
+    /**
+     * Migrates all consumers of the given anonymous org to the target org.
+     *
+     * @param originOwnerKey key of the source anonymous org
+     * @param destinationOwnerKey key of the destination org
+     */
+    public void migrate(String originOwnerKey, String destinationOwnerKey) {
+        Owner originOwner = this.ownerCurator.getByKey(originOwnerKey);
+        Owner destinationOwner = this.ownerCurator.getByKey(destinationOwnerKey);
+
+        List<String> consumerIds = this.ownerCurator.getConsumerIds(originOwner).list();
+        for (List<String> consumerIdsBlock : Iterables.partition(consumerIds, this.batchSize)) {
+            try {
+                this.ownerCurator.transactional(args -> {
+                    migrateBatch(consumerIdsBlock, destinationOwner);
+                    return null;
+                }).execute();
+            }
+            catch (Exception e) {
+                log.warn("Failed to migrate a batch of consumers.", e);
+                // Retry later
```

Comment:
I think if at least one transaction fails, we want to fail the job so that it is retried later

Reply:
clarification after discussion: we should still try to work through all the batches, but if any of them fail, we should mark it with a flag, so we fail the job at the very end

Reply:
Done.